internal/controller/statefulsetpool_controller.go

```diff
@@ -0,0 +1,462 @@
+}
+
+func (r *StatefulsetPoolReconciler) statusConditionController(ctx context.Context, statefulsetPool *kwoksigsv1beta1.StatefulsetPool, condition metav1.Condition) error {
+	meta.SetStatusCondition(&statefulsetPool.Status.Conditions, condition)
+	return r.Status().Update(ctx, statefulsetPool)
+}
+
+func (r *StatefulsetPoolReconciler) addFinalizer(ctx context.Context, statefulsetPool *kwoksigsv1beta1.StatefulsetPool) error {
+	controllerutil.AddFinalizer(statefulsetPool, controllerFinalizer)
+	return r.Update(ctx, statefulsetPool)
+}
+
+func (r *StatefulsetPoolReconciler) deleteFinalizer(ctx context.Context, statefulsetPool *kwoksigsv1beta1.StatefulsetPool) error {
+	controllerutil.RemoveFinalizer(statefulsetPool, controllerFinalizer)
+	return r.Update(ctx, statefulsetPool)
+}
+
+func (r *StatefulsetPoolReconciler) getStatefulset(ctx context.Context, statefulsetPool *kwoksigsv1beta1.StatefulsetPool) (*appsv1.StatefulSet, error) {
+	// get Statefulset with the name of statefulsetPool name
+	Statefulset := &appsv1.StatefulSet{}
+	err := r.Get(ctx, client.ObjectKey{
+		Namespace: statefulsetPool.Namespace,
+		Name:      statefulsetPool.Name,
+	}, Statefulset)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return Statefulset, nil
+}
+
+func (r *StatefulsetPoolReconciler) getPVCList(ctx context.Context, statefulsetPool *kwoksigsv1beta1.StatefulsetPool) (*corev1.PersistentVolumeClaimList, error) {
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	err := r.List(ctx, pvcList, client.MatchingLabels{controllerLabel: statefulsetPool.Name})
+	if err != nil {
+		return nil, err
+	}
+	return pvcList, nil
+}
+
+func (r *StatefulsetPoolReconciler) getPVList(ctx context.Context, statefulsetPool *kwoksigsv1beta1.StatefulsetPool) (*corev1.PersistentVolumeList, error) {
+	pvList := &corev1.PersistentVolumeList{}
+	err := r.List(ctx, pvList, client.MatchingLabels{controllerLabel: statefulsetPool.Name})
+	if err != nil {
+		return nil, err
+	}
+	return pvList, nil
+}
+
+// update Statefulset
+func (r *StatefulsetPoolReconciler) updateStatefulset(ctx context.Context, statefulsetPool *kwoksigsv1beta1.StatefulsetPool) error {
+	// get the Statefulset spec from the cluster
+	Statefulset, err := r.getStatefulset(ctx, statefulsetPool)
+	replicas := statefulsetPool.Spec.StatefulsetTemplate.Spec.Replicas
+	log.Log.Info("Updating Statefulset", "Statefulset", Statefulset)
+	if err != nil {
+		return err
+	}
+	pvList, err := r.getPVList(ctx, statefulsetPool)
+	if err != nil {
+		return err
+	}
+	Statefulset.Spec.Replicas = replicas
+	Statefulset.Spec.Template.Spec.Containers = statefulsetPool.Spec.StatefulsetTemplate.Spec.Template.Spec.Containers
+	// scale up pv in case of replicas change
+	if statefulsetPool.Spec.CreatePV && statefulsetPool.Spec.StatefulsetTemplate.Spec.VolumeClaimTemplates != nil && int(*replicas) > int(len(pvList.Items)) {
+		storageClassName, err := r.getStorageClassName(ctx)
```

Comment:
add logical that if storageclass name meantion in statefulset it will take it , if not will pull the getDeafultStorageClassName func